app/services/data_migrations/restore_deleted_courses.rb

```diff
@@ -0,0 +1,58 @@
+        58058,55,Mathematics,H902,2023-10-02 23:11:10.880577,2023-10-02 23:11:10.925319,Secondary,f,2024,F,,2024-08-31 23:00:00,,QTS full time teaching apprenticeship,468,apprenticeship,14 to 19,"[""qts""]",TA,f,f21f126b-211b-4dcd-8a4c-52b1f255100c,not_required,,f,f,f,f,f,,2023-10-09 23:00:00,,,,,f,f,0
+        58703,2033,Primary,S435,2023-10-02 23:11:16.625958,2023-10-02 23:11:16.660774,Primary,f,2024,F,,2024-08-31 23:00:00,,QTS full time teaching apprenticeship,,apprenticeship,3 to 7,"[""qts""]",TA,f,c6aa2897-b092-4c00-8ba5-d83892624e38,two_two,,t,t,t,t,t,"If you do not hold a GCSE in English, maths or science at grade 4 (C) or above or equivalent you can apply to take an equivalency test. We accept equivalency tests from:
+
+        * [A*Star Teachers](www.astarteachers.co.uk)
+        * [Equivalency Testing](www.equivalencytesting.co.uk)
+        * [Maths Made Easy](www.mathsmadeeasy.co.uk)",2023-10-09 23:00:00,,,,,f,f,0
+        60253,29,Physical Education,368Q,2023-10-02 23:11:28.749743,2023-10-02 23:11:28.772544,Secondary,f,2024,F,,2024-08-31 23:00:00,,PGCE with QTS full time,30,fee,11 to 16,"[""qts"", ""pgce""]",SD,f,f76bda35-b5f5-49bf-9e14-2eafeebcf40f,two_two,,t,t,t,t,,"",2023-10-09 23:00:00,,,,,f,f,0
+        60784,475,Primary (Special Educational Needs),2HP5,2023-10-02 23:14:11.375273,2023-10-02 23:14:11.43646,Primary,f,2024,F,,2024-08-31 23:00:00,,PGCE with QTS full time,1071,fee,5 to 11,"[""qts"", ""pgce""]",SD,f,ce437390-2971-412d-bdf6-d7ca31bb91b4,two_two,,t,t,t,t,t,"Any equivalency tests completed must be equivalent to GCSE level. We do not accept adult learning courses as these do not offer the breadth of knowledge that the GCSE curriculums cover.
+
+        We recommend using Edgehill University or those offered by Prestolee SCITT. Links to both of these websites are below:
+
+        https://www.edgehill.ac.uk/study/equivalency/?tab=standalone-tests
+
+        https://www.prestoleescitt.co.uk/Equivalency-Testing/",2023-10-09 23:00:00,,,,,f,f,0
+        66304,1417,Primary,2YRM,2023-10-02 23:20:20.991307,2023-10-02 23:20:21.067816,Primary,f,2024,F,,2024-08-31 23:00:00,,QTS full time,1033,fee,5 to 11,"[""qts""]",SD,f,b2483b4f-c3c6-4768-9f64-ef9c602c1d43,two_two,,t,t,t,t,t,"A candidate who does not hold a GCSE at the required grade can either retake a GCSE if time permits, or take
+        an equivalency test. At present, we can only accept this from https://www.equivalencytesting.com/.
+        ",2023-10-09 23:00:00,,,,,f,f,0
+      CSV
+
+      Course.insert_all(CSV.parse(courses_csv, headers: true).map(&:to_h))
```

Comment:
Is it worth wrapping in a transaction incase any part of the data fails to insert?

Is it also worth adding some logging?

Reply:
Sorted in ee843a44

Reply:
Transaction addressed, logging is already automatically from the insert_all